posts/2018-02-09-serverless-and-graphql-in-love.md

```diff
@@ -0,0 +1,611 @@
+3. Describe your API with types that map your schema to existing backend.
+
+Thousands of companies are now using GraphQL in production with the help of open source frameworks built by Facebook, Apollo, and Graphcool. In fact, that [Starbucks announcement](https://twitter.com/davidbrunelle/status/960946257643454464) last week made my morning coffee taste even better. 😉
+
+![](https://user-images.githubusercontent.com/1587005/36067514-503a05b4-0e73-11e8-9b40-946c5398f4b5.png "Danielle's slide from Serverless and GraphQL meetup at Glassdoor, Jan 29, 2018")
+
+Now, very reasonably, you are probably thinking, “Yeah, okay, Facebook is one thing; they have a giant engineering team. But for me, having only one API endpoint is too risky. What if it goes down? How do I handle that much load? What about security?”
+
+You are absolutely correct: with one HTTP endpoint, you need be be entirely sure that endpoint never goes down and that it scales on demand.
+
+That’s where serverless comes in.
+
+## What is Serverless?
+
+Serverless has gained popularity over last few years, primarily because it gives developers flexibility.
+
+With Serverless comes the following:
+1. No server management (no need to manage any form of machine)
+2. Pay-per-execution (never pay for idle)
+3. Authentication and authorization at scale
+4. Function as a unit of application logic
+
+## What makes Serverless and GraphQL such a great fit?
+
+When moving to GraphQL, you suddenly rely on one HTTP endpoint to connect your clients to your backend services. Once you do decide to do that, you want this one HTTP endpoint to be: reliable, fast, auto-scaling, and have a small attack vector regarding security.
+
+*All* these properties are fulfilled by a single AWS Lambda function in combination with API Gateway. It’s just a great fit! 
+
+**Note:** Check out [Jared Short's Serverless + GraphQL post](https://www.trek10.com/blog/a-look-at-serverless-graphql/) to get an even better understanding of this relationship 😎
+
+In sum, powering your GraphQL endpoint with a serverless backend solves scaling and availability concerns outright, and it gives you a big leg up on security. It’s not even that much code or configuration.
+
+It takes only a few minutes to get to a production-ready setup, which we're about to dive into, right now.
+
+## Serverless-GraphQL repository
+
+With the shiny new [Serverless and GraphQL Repository](https://github.com/serverless/serverless-graphql), it’s incredibly straightforward to get your HTTP endpoint up and running.
+
+![alt text](https://user-images.githubusercontent.com/1587005/36035218-1c06763c-0d6b-11e8-996b-996243b0975f.png "Serverless and GraphQL Architecture")
+
+The repository comes in two flavors: API Gateway and Lambda backend, or AppSync backend. (And we're currently working on adding more backend integrations, including [GraphCool Prisma](https://github.com/graphcool/prisma), Druid, MongoDB, and AWS Neptune.)
```

Comment:
`API Gateway and Lambda backend` --> `AWS API Gateway + Lambda backend` since the `+` shows that it's a combination of both services, rather than a normal sentence.